server/mdm/apple/profile_verifier.go

```diff
@@ -99,8 +99,11 @@ func VerifyHostMDMProfiles(ctx context.Context, ds fleet.ProfileVerificationStor
 // the MDM protocol and updates the verification status in the datastore. It is intended to be
 // called by the Fleet MDM checkin and command service install profile request handler.
 func HandleHostMDMProfileInstallResult(ctx context.Context, ds fleet.ProfileVerificationStore, hostUUID string, cmdUUID string, status *fleet.MDMDeliveryStatus, detail string) error {
-	host := &fleet.Host{UUID: hostUUID, Platform: "darwin"}
 	if status != nil && *status == fleet.MDMDeliveryFailed {
+		// Here we set the host.Platform to "darwin" but it applies to iOS/iPadOS too.
+		// The logic in GetHostMDMProfileRetryCountByCommandUUID and UpdateHostMDMProfilesVerification
+		// is the exact same when platform is "darwin", "ios" or "ipados".
+		host := &fleet.Host{UUID: hostUUID, Platform: "darwin"}
```

Comment:
Maybe we need a concept of a super-platform, like "linux" and "ubuntu", etc.
In this case, it would be "apple" and "darwin", or something similar.

Reply:
Yeah, on some places we use "darwin" to represent "macOS" and in others to represent "apple".